### ISSUE TITLE
enable session snapshot and periodic backup for multiple instances of notepad++

### ISSUE THREAD

**Issue by madscijr** [notepad-plus-plus/notepad-plus-plus#12533]
### Description of the Issue

Notepad++ has the option to “enable session snapshot and periodic backup”, where if you cleanly exit Notepad++ without saving, it will remember any changes that haven’t been saved, and remember which files are open. When you relaunch Notepad++, it will reload the file, even with the unsaved changes. This usually even works when you shut down Windows (or Windows auto-restarts) with Notepad++ open. However, if the user has multiple instances of Notepad++ open, this does not work for the additional instances. Currently Notepad++ only allows one instance of the application to have access to the session file. 

### Steps to Reproduce the Issue

1. Open 2 instances of Notepad++ with a different set of files in each (including new files that are not saved at all yet). Type some text in the new windows.
2. Restart Windows.
3. Re-open Notepad++. The 2nd instance you had open does not re-open, and the unsaved files in it are not recoverable. 

### Expected Behavior

Expect Notepad++ to recover all instances.

### Actual Behavior
Notepad++ only recovers the first instance.

### Debug Information

(N/A - does not apply because this is for a new feature)

This would help a use case where Windows auto-updates and auto-reboots (something many people have no control over at their workplace due to security settings) while the user is away from their computer, and had multiple instances of Notepad++ open.

**Comment by pdelaage** [comment]
Would appreciate as well that the "save(/restore) session" features be able to include all Notepad++ Windows that are (were) open. In the meantime, I have to perform a save session by hand on each of the open windows to be able to restore them (eg in case of a system crash) one by one, by hand.

**Comment by bwims** [comment]
I consider this a bug.  When I start a new tab in the 1st instance, it creates a backup of it in the 7 second time frame.
When I create a new instance, firstly I notice that the "Enable session snapshot..." has been unchecked, but secondly, even if then checked,  no backup tabs are created for that instance.

A bug could be called an undocumented feature.  If this behaviour is intended, please point out where.  Otherwise, please attach the "Bug" label to this issue.

By the way, I am running the latest version:

Notepad++ v8.4.8   (64-bit)
Build time : Dec 24 2022 - 19:40:27
Path : C:\Program Files\Notepad++\notepad++.exe
Command Line : 
Admin mode : OFF
Local Conf mode : OFF
Cloud Config : OFF
OS Name : Windows 10 Pro (64-bit) 
OS Version : 22H2
OS Build : 19045.2846
Current ANSI codepage : 1252
Plugins : 
    MarkdownViewerPlusPlus (0.8.2)
    mimeTools (2.9)
    NppConverter (4.5)
    NppExport (0.4)
    NPPJSONViewer (1.40)



Thanks

B.

**Comment by QGtKMlLz** [comment]
I STRONGLY agree with, sometimes an exit is caused by a crash and then I lose have the files I am working on.  

We don't even have an option to backup multiple session/instance files so we could manually backup/restore.

**Comment by bwims** [comment]
Bump!  

Surely attaching a bug label is not too onerous?

> A bug could be called an undocumented feature. If this behaviour is intended, please point out where. Otherwise, please attach the "Bug" label to this issue.
>
